src/test/java/org/cactoos/list/NoNullsTest.java

```diff
@@ -167,7 +166,6 @@ public void removeThrowsErrorForImmutableListIterator() {
                 return 0;
             },
             new Throws<>(
-                "Iterator is read-only and doesn't allow removing items",
```

Comment:
@fanifieiev why remove those parts of the tests?

Reply:
@victornoel If you look at the test and the following stacktrace, you can see that the exception is thrown by the collection returned by `Collections.unmodifiableList`, that is why it throws exception with no message. 
![image](https://user-images.githubusercontent.com/3167940/68537757-85bc0d80-0337-11ea-9b32-dddb30302de9.png)

Reply:
@fanifieiev I better understand, thank you, but why is `ListIteratorNoNulls` based on an immutable version of `ListIterator`? I think that it is coherent we the changes we are making that `ListIteratorNoNulls` is to be only concerned with nulls, not with mutability. Please fix it or add a todo to continue the work.

Reply:
@victornoel I did not catch your concern. The  [ListIteratorOf](https://github.com/yegor256/cactoos/pull/1231/commits/52758f3b87a55249bdfdff4e8ae76f7847bb7285#diff-938af5d32ea1b9126e592148160ddbe9) I changed  is not based on the immutable version of `ListIterator`.
Could you provide details regarding you concern?

Reply:
I'm talking about `ListIteratorNoNulls`: those tests on which we are currently commenting seem to validate that it is immutable but I don't think it is in the spirit of the changes that are being done to the `ListIterator` type hierarchy in this PR: either change `ListIteratorNoNulls` to be coherent with the changes you are currently making, or add a todo in order for someone to do it.

Reply:
@victornoel The test we are currently commenting on is [tests](https://github.com/yegor256/cactoos/pull/1231/files/52758f3b87a55249bdfdff4e8ae76f7847bb7285#diff-f3ddcee3c2a90c165c537d1734be8fdc).
If yes, then those tests are not using 'ListIteratorNoNulls', they use 'NoNulls' list and the reason why this behavior is immutable, is that 'ListOf' uses 'Collections.unmodifiableList'.
So, there is nothing related to 'ListIteratorNoNulls'.

Reply:
@fanifieiev I understand, ok. But in any case, this is related to the work you are doing right now (that `ListIteratorOf` should not enforce immutability) and I think it would be relevant to continue this work by adding a todo. So in this case that `list.NoNulls` nor `ListOf`, nor `ListIterableNoNulls` should not enforce immutability. It's not because we are working on a small tasks that we shouldn't think about the bigger picture, this is the difference between chaos and iterative/incremental development.

Reply:
@fanifieiev as a side comment: as the DEV working on this, you are the best person to evaluate what are the next changes to continue implemen the vision that the current task you are doing is promoting.

Reply:
@victornoel Now your concern is clear to me.
What if I make changes to all mentioned classes(list.NoNulls nor ListOf, nor ListIterableNoNulls should not enforce immutability) in scope of this PR ?
It will be faster, than creating a todo and waiting  for someone fix/review.

Reply:
@fanifieiev it's good for me, your choice :)

Reply:
@victornoel @paulodamaso A lot of list implementations in the `org.cactoos.list` package construct `Collections.unmodifiableList` which makes the lists immutable.
In my opinion we do not have to enforce them be immutable, but rather mutable and each of them bears its own specific responsibility. 
If someone wants to construct, for example, an immutable 'org.cactoos.list.Sorted' list, he/she should decorate it with `org.cactoos.list.Immutable`: `new Immutable<>(new Sorted(10, 2, 3, 5))`.
So, using this approach, we can provide API, that can be used in both mutable/immutable cases.
I am not 100% sure this idea is prefect, as the `Collections.unmodifableList` was added by @yegor256 in the above mentioned lists.
Please share you thoughts.

Reply:
@fanifieiev to me you are rephrasing what I already said and what was decided by ARC with the accord of @yegor256 previously in #898. You just need to create the todos et and let's get on with it.

Reply:
@victornoel I can fix that in scope of current PR if this is acceptable.

Reply:
@fanifieiev I think we spent enough time on this PR, please add todos and let ARC validate them by merging.

Reply:
@fanifieiev one todo is enough, no need to multiplicate them btw :)

Reply:
@victornoel There is one thing that concerns me.
Let's say we will make those mentioned classes mutable. Let's take `Sorted` as an example:
`List<Integer> sorted = new Sorted<>(2, 1, 0, 5);` //sorted list is now sorted
Let's add one more element:
`sorted.add(6);`
//sorted list is not sorted anymore
You got the idea of my concern?

Reply:
@fanifieiev that's why you should leave a todo and let the next job performer think about this problem. Maybe we will conclude that `Sorted` can't be mutable, but that's out of scope of the present issue. Personally I believe the real problem is that it's not relevant to decorate collection to sort them, but that's only my opinion ;) Also, nothing prevent to implement Sorted using a Java ordered list and implement insertion in a sorted manner.

Reply:
@victornoel Ok, I have just pushed one more commit with 'todo'. Please have a look.